src/IO.Ably.Net45/WebsocketTransport.cs

```
@@ -189,8 +189,10 @@ private void socket_DataReceived(object sender, DataReceivedEventArgs e)
             {
                 try
                 {
+#if MSGPACK
                     var message = MsgPackHelper.DeserialiseMsgPackObject(e.Data).ToString();
                     Logger.Debug("Websocket data message received. Raw: " + message);
+#endif
```

same as above. ☝️ Just add a log message to say that's unexpected when the binary protocol is off.

---

If MSGPACK is disabled then e.Data should be JSON, so could I do something like below?

Logger.Debug("Websocket data message received. Raw: " + Encoding.UTF8.GetString(e.Data));

---

I think we should raise an exception.  This code path should never happen when MsgPack is disabled.

---

@withakay if you have useBinary false you will never receive binary data from the socket. You can safely throw an exception